cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -481,7 +481,10 @@ class HydratorPlusPlusTopPanelCtrl {
   }
 
   startOrStopPreview() {
-    if (this.doesPreviewHaveEmptyMacros) {
+    if(this.previewRunning){
+      this.stopPreview();
+    }
+    else if (this.doesPreviewHaveEmptyMacros) {
       this.doStartOrStopPreview();
```

Comment:
this function signature doesn't make sense anymore

Reply:
refactored the start stop functions